datatypes/src/raster/operations/from_index_fn.rs

```diff
@@ -103,9 +103,8 @@ where
             })
             .unzip();
 
-        let inner_grid = Grid::new(g.clone(), out_data).expect("grid shape was invalid before");
-        let validity_mask =
-            Grid::new(g.clone(), out_validity).expect("grid shape was invalid before");
+        let inner_grid = Grid::new(g.clone(), out_data).expect("grid shape should be valid");
+        let validity_mask = Grid::new(g.clone(), out_validity).expect("grid shape should be valid");
```

Comment:
well the same argument. It should be valid other wise it would fail. However, the reason why we expect it to work is that the same case / composition of shapes was used before and therefore it can not fail here..

Reply:
0bc74c57b981a9d1509efdf2b3c5b412f68e404f